network/phonebook.go

```diff
@@ -47,50 +47,78 @@ type Phonebook interface {
 	// UpdateConnectionTime will update the provisional connection time.
 	// Returns true of the addr was in the phonebook
 	UpdateConnectionTime(addr string, provisionalTime time.Time) bool
+
+	// ReplacePeerList merges a set of addresses with that passed in for networkName
+	// new entries in dnsAddresses are being added
+	// existing items that aren't included in dnsAddresses are being removed
+	// matching entries don't change
+	ReplacePeerList(dnsAddresses []string, networkName string)
+
+	// ExtendPeerList adds unique addresses to this set of addresses
+	ExtendPeerList(more []string, networkName string)
 }
 
-type phonebookData struct {
+type phonebookEntryData struct {
 	retryAfter            time.Time
 	recentConnectionTimes []time.Time
+	networkNames          map[string]bool
 }
 
-// phonebookEntries holds the server connection configuration values
-// and the list of request times withing the time window for each
+func makePhonebookEntryData(networkName string) phonebookEntryData {
+	pbData := phonebookEntryData{
+		networkNames:          make(map[string]bool),
+		recentConnectionTimes: make([]time.Time, 0),
+	}
+	pbData.networkNames[networkName] = true
+	return pbData
+}
+
+// phonebookImpl holds the server connection configuration values
+// and the list of request times within the time window for each
 // address.
-type phonebookEntries struct {
+type phonebookImpl struct {
```

Comment:
how about just a "phonebook"? lowercase and no Impl suffix?

Reply:
@tsachiherman ?

Reply:
I think that after eliminating the phonebook from any external source ( i.e. node ), you'll find that the only consumer for the phonebook is the wsNetwork. When we get to that point, it would make sense to delete the interface and just use the rename `phonebookImpl` -> `phonebook`. But we're not there yet.

Reply:
This point will be resolved in a future PR.